data_pipeline/scraper/scraper.py

```diff
@@ -19,65 +19,101 @@
 from time import sleep
 from dircreator import DirCreator
 
+# Default website wait time.
+DEFAULT_SLEEP_TIME = 5
 
-class Scraper():
-    def __init__(self):
-        # create data folder in current directory to store files
-        new_dir = DirCreator()
-        new_dir.createFolder()
-        self.path_dir = new_dir.changeDirectory()
-
-        # time to sleep()
-        self.s = 2
 
-        self.drivers = []
-        self.driver = None
-
-    def scrape(self):
-        # create drivers for each page on site
-        self.__initDrivers()
+class SjcWebsite():
+    """This class represents an interface to interact with the elements on the SJC website.
 
-        error_dialog = False
-        end = False
-        for i in range(6, 9):
-            # This code is from IterForms, it downloads the forms on a page.
-            sleep(self.s + 2)
-            forms = self.driver.find_elements_by_xpath(
-                '//a[@class="dxbButton_Glass dxgvCommandColumnItem_Glass dxgv__cci dxbButtonSys"]')
-
-            for ind, form in enumerate(forms):
-                sleep(self.s)
-                if not error_dialog:
-                    for x in range(i):
-                        othername = self.driver.find_element_by_css_selector('[alt="Next"]').find_element_by_xpath('..')
-                        othername.click()
-                        sleep(self.s + 1)
-                error_dialog = False
-
-                forms = self.driver.find_elements_by_xpath(
-                    '//a[@class="dxbButton_Glass dxgvCommandColumnItem_Glass dxgv__cci dxbButtonSys"]')
-                forms[ind].click()
-
-                sleep(self.s + 1)
+    It abstracts away details like "what CSS class does a specific button have", and implementation details of the
+    website, while still requiring users to understand how to navigate around the website.
+    """
+    def navigateToSearchPage(self, driver):
+        driver.get(
+        "https://www.southtechhosting.com/SanJoseCity/CampaignDocsWebRetrieval/Search/SearchByElection.aspx")
+        WebDriverWait(driver, 10).until(
+            EC.presence_of_element_located((By.ID, 'ctl00_DefaultContent_ASPxRoundPanel1_btnFindFilers_CD')))
 
-                # TODO(walek): Update this to use contains
-                if self.driver.find_elements_by_xpath('//*[@id="ctl00_GridContent_popupCantContinueDialog_Button1"]'):
-                    self.driver.find_elements_by_xpath('//*[@id="ctl00_GridContent_popupCantContinueDialog_Button1"]')[0].click()
-                    error_dialog = True
-                    continue
+        # Search, which will load up the content on the page.
+        # Search terms are left blank, indicating "all content".
+        driver.find_element_by_xpath('//*[@id="ctl00_DefaultContent_ASPxRoundPanel1_btnFindFilers_CD"]').click()
```

Comment:
can we set these ids to better var names that are human readable and make more sense? helps navigate the scrapper even in code

Reply:
same with any others

Reply:
The ID's are generated as part of the city website which unfortunately we do not control.  I did abstract these out into variables though to try to make it more readable at least within this file. Thanks for this input.